trussworks/react-uswds
# chore(deps): upgrade node and allow any version newer than 16.20.0

# Summary

## Related Issues or PRs

Bump version of node in `.node-version` to 18.15.0. 18.16.0 isn't available in `nodenv` yet so didn't add it. I also changed the package json to allow any version newer than the minimum of 16.20.0 to allow folks to try this out with node 19.x, similar to #2329.

I updated the README to direct folks to the files that store the versions so that we don't have to updated it in multiple places.

This PR also drops support for node 14 as it is end of life 2023-04-30, which will be around when we release the next version. See https://nodejs.dev/en/about/releases/

Since node 14 is being dropped we will need to update the branch protections before merging this PR.

## How To Test

Ensure everything seems to run under node 18.

```sh
yarn lint
yarn test
yarn storybook
```
